LeaderboardBackend/Services/Impl/RunService.cs

```diff
@@ -132,12 +133,27 @@ public async Task<UpdateRunResult> UpdateRun(User user, Guid id, UpdateRunReques
         {
             case UserRole.Confirmed:
             {
-                if (run.UserId == user.Id)
+                if (run.UserId != user.Id)
                 {
-                    break;
+                    return new UserDoesNotOwnRun();
                 }
 
-                return new BadRole();
+                if (run.DeletedAt != null)
+                {
+                    return new AlreadyDeleted();
+                }
+
+                if (run.Category.DeletedAt != null)
+                {
+                    return new AlreadyDeleted(typeof(Category));
+                }
+
+                if (run.Category.Leaderboard!.DeletedAt != null)
+                {
+                    return new AlreadyDeleted(typeof(Leaderboard));
+                }
```

Comment:
It might be simpler to just have different results for these cases: `Deleted`, `CategoryDeleted`, `LeaderboardDeleted` for example.

Reply:
I really don't think so. We won't change anything on the controller's end if we do define these new results. All we're doing is creating extra one-use-case results.